var/spack/repos/builtin/packages/erlang/package.py

```diff
@@ -14,10 +14,12 @@ class Erlang(AutotoolsPackage):
     """
 
     homepage = "https://erlang.org/"
-    url = "https://erlang.org/download/otp_src_23.3.tar.gz"
+    url = "https://github.com/erlang/otp/releases/download/OTP-23.3/otp_src_23.3.tar.gz"
 
-    license("Apache-2.0")
+    license("Apache-2.0", checked_by="wdconinc")
 
+    version("27.0.1", sha256="26d894e2f0dda9d13560af08ea589afc01569df6b5486e565beb5accb99c9cf4")
+    version("26.2.5.2", sha256="e49708cf1f602863e394869af48df4abcb39e3633b96cb4babde3ee7aa724872")
```

Comment:
Apparently old version shasums changed with the url:
```
==> Fetching https://github.com/erlang/otp/releases/download/OTP-27.0.1/otp_src_27.0.1.tar.gz
==> Fetching https://github.com/erlang/otp/releases/download/OTP-26.0/otp_src_26.0.tar.gz

    version("27.0.1", sha256="26d894e2f0dda9d13560af08ea589afc01569df6b5486e565beb5accb99c9cf4")
    version("26.0", sha256="3ff3c53d7ef9a45b5720e95b8756269c1a1b58eb51accc992ca97522fdb234d4")
```

Reply:
I'd have thought that for a project the size of erlang they would not have different released archives in multiple places... :-(

Diffs are various and appear innocuous, e.g.
```diff
168c168
< <timestamp>Generated by EDoc, May 15 2023, 21:36:54.</timestamp>
---
> <timestamp>Generated by EDoc, May 16 2023, 09:52:45.</timestamp>
```

It seems that the listings on e.g. https://www.erlang.org/patches/otp-24.3.4.17, going back as far as v17 https://www.erlang.org/patches/otp-17.5.1 have the authoritative source code link pointing to GitHub. References to erlang.org/download/ are almost exclusively in _news/*.md announcements, including some [recent](https://github.com/erlang/erlang-org/blob/master/_news/169.md?plain=1#L240): https://github.com/search?q=repo%3Aerlang%2Ferlang-org+%22https%3A%2F%2Ferlang.org%2Fdownload%22&type=code

Not the first time that versions do not make it to erlang.org/download, e.g. https://github.com/erlang/erlang-org/issues/94 where the suggestion was to download from GitHub.

So, my proposal: make the GitHub url authoritative and replace the hashes of older version. Thoughts?